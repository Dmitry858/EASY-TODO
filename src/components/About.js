import React from 'react';
import { connect } from 'react-redux';
import Topbar from './Topbar';

const About = (props) => {

    return (
        <React.Fragment>
            {(props.type === 'inner') && <Topbar />}

            <div className={(props.type === 'inner') ? 'content content-inner' : 'content'}>
                <div className="container">
                    <div className="title-wrap">
                        <h1>О сервисе</h1>
                    </div>
                    <p>Далеко-далеко за словесными горами в стране гласных и согласных живут рыбные тексты. Деревни последний грамматики языком скатился большого! Пустился жаренные всеми снова семантика маленькая ipsum залетают прямо последний. Пустился от всех не проектах! Имени не необходимыми lorem, свой пояс сбить реторический все! Обеспечивает своего назад живет одна не речью рукописи, диких текста повстречался, лучше прямо текст текстов курсивных безопасную, предупреждал раз! Напоивший, единственное.</p>
                </div>
            </div>
        </React.Fragment>
    );

};

export default connect(state => state)(About);