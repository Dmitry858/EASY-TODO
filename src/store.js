import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import tasksStoreConfig from './reducers/tasks';

const createReducer = (initialState, handlers) => {
    return (state = initialState, action) => {
        return (handlers[action.type] && handlers[action.type](state, action)) || state;
    };
};

const tasksReducers = createReducer(tasksStoreConfig.initialState, tasksStoreConfig.actions);

const rootReducer = combineReducers({
    tasks: tasksReducers,
});

export default createStore(rootReducer, {}, applyMiddleware(thunk));