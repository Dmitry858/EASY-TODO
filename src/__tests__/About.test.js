import React from 'react';
import ReactDOM from 'react-dom';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import qhistory from 'qhistory';
import { stringify, parse } from 'qs';
import About from '../components/About';
import { Provider } from 'react-redux';
import store from '../store.js';
import { render } from '@testing-library/react';

describe('About testing', () => {
    const history = qhistory(
        createBrowserHistory({}),
        stringify,
        parse
    );

    const type = 'inner';

    const AboutWithProvider = () => {
        return (
            <Provider store={store}>
                <Router history={history}>
                    <About type={type} />
                </Router>
            </Provider>  
        );      
    };

    it('About render without crashing', () => {
        const div = document.createElement('div');
      
        ReactDOM.render(
            <AboutWithProvider />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('About matches the snapshot', () => {
        const { container } = render(<AboutWithProvider />);
        expect(container).toMatchSnapshot();
    });

});
