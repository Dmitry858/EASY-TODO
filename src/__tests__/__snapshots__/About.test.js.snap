// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About testing About matches the snapshot 1`] = `
<div>
  <div
    class="topbar"
  >
    <span>
      <a
        class="notice-link"
        href="#"
      >
        <i
          aria-hidden="true"
          class="fa fa-bell"
        />
      </a>
    </span>
    <a
      href="#"
    >
      <i
        aria-hidden="true"
        class="fa fa-sign-out"
      />
    </a>
  </div>
  <div
    class="content content-inner"
  >
    <div
      class="container"
    >
      <div
        class="title-wrap"
      >
        <h1>
          О сервисе
        </h1>
      </div>
      <p>
        Далеко-далеко за словесными горами в стране гласных и согласных живут рыбные тексты. Деревни последний грамматики языком скатился большого! Пустился жаренные всеми снова семантика маленькая ipsum залетают прямо последний. Пустился от всех не проектах! Имени не необходимыми lorem, свой пояс сбить реторический все! Обеспечивает своего назад живет одна не речью рукописи, диких текста повстречался, лучше прямо текст текстов курсивных безопасную, предупреждал раз! Напоивший, единственное.
      </p>
    </div>
  </div>
</div>
`;
